### ISSUE TITLE
Hybrid Runbook Worker registration

### ISSUE THREAD

**Issue by itbland** [MicrosoftLearning/10992-IntegratingOn-PremisesInfrastructurewithMicrosoft-Azure#52]
In Lab 07, the installation of the Hybrid Runbook worker is now accomplished with a PowerShell script installed from the PowerShell Gallery. However, the install fails on MIA-CL1 in the lab. It is possible to install the 1.4 version of the script, but the registration fails. I assume there is a dependency on updating PowerShell, or possibly a missing OS update causing the problem. The script can be extracted and run manually, but requires many steps to do so.
[Automated-Deployment-HRW](https://docs.microsoft.com/en-us/azure/automation/automation-windows-hrw-install#automated-deployment)
(NOTE: The manual deployment is no longer available)

**Comment by polichtm** [comment]
What errors are you seeing when running the script?
rgds
Marcin

**Comment by itbland** [comment]
When running:
Install-Script -Name New-OnPremiseHybridWorker

MIA-CL1 returns this error:

PackageManagement\Install-Package : Script
'C:\Users\Student\AppData\Local\Temp\1593946071\New-OnPremiseHybridWorker\New-OnPremiseHybridWorker.ps1' is missing
required metadata properties. Verify that the script file has Version, Guid, Description and Author properties. You
can use the Update-ScriptFileInfo or New-ScriptFileInfo cmdlet to add or update the PSScriptInfo to the script file.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:3389 char:21
+ ...          $null = PackageManagement\Install-Package @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (C:\Users\Studen...ybridWorker.ps1:String) [Install-Package], Exception
    + FullyQualifiedErrorId : MissingRequiredPSScriptInfoProperties,Test-ScriptFileInfo,Microsoft.PowerShell.PackageMa
   nagement.Cmdlets.InstallPackage


I do not see this error when installing on an up to date machine.

-Eric
________________________________

**Comment by itbland** [comment]
Marcin,
It turns out that we _CAN_ still do the manual HRW installation, we just **need to add an Automation solution** to the workspace before we download the agent. The step we need is: 

> To add the Automation Azure Monitor logs solution to your workspace, run the following PowerShell.
**Set-AzureRmOperationalInsightsIntelligencePack -ResourceGroupName <logAnalyticsResourceGroup> -WorkspaceName <LogAnalyticsWorkspaceName> -IntelligencePackName "AzureAutomation" -Enabled $true**

[Deploy a Windows Hybrid Runbook Worker](https://docs.microsoft.com/en-us/azure/automation/automation-windows-hrw-install#feedback)

**Comment by polichtm** [comment]
Good to see. I'll update the instructions accordingly 
rgds
Marcin

**Comment by polichtm** [comment]
updated

**Comment by itbland** [comment]
Marcin,
The original error I encountered here in #52 was due to missing steps in the Learnondemand copy of the instructions. The PowerShell command to add the HRW package to the Workspace (Exercise 2, Task 1, Step 3) is actually not needed because we add the HRW in Exercise 1, Task 3.  Sorry for the confusion, LOD steps are mostly in sync with these steps, but not always. (I verified that skipping that one PS command works.)
-Eric
